### PR TITLE
Switch to our fork of goja - sobek

### PR DESCRIPTION
**Pull request by mstoykov**
## What?

Move to sobek from goja - currently only aliasing types over from goja.

## Why?

This is part of #3772 and #3773.

The PR also doesn't try to fix every other reference of goja - a bunch of them are docs that will break at this point, and will likely need to be updated over time. Likely after sobek is an actual fork of goja.

This is marked as breaking change mostly because I do expect it will still break something, even if not for most users.

## Checklist

- [x] I have performed a self-review of my code.
- [x] I have added tests for my changes.
- [x] I have run linter locally (`make lint`) and all checks pass.
- [x] I have run tests locally (`make tests`) and all tests pass.
- [ ] I have commented on my code, particularly in hard-to-understand areas.

### CODE REVIEW

File: js/modules/k6/experimental/streams/readable_streams.go [hunk 1]
```diff
@@ -62,9 +62,10 @@ func (stream *ReadableStream) Cancel(reason goja.Value) *goja.Promise {
 // GetReader implements the [getReader] operation.
 //
 // [getReader]: https://streams.spec.whatwg.org/#rs-get-reader
-func (stream *ReadableStream) GetReader(options *goja.Object) goja.Value {
+func (stream *ReadableStream) GetReader(options *sobek.Object) sobek.Value {
 	// 1. If options["mode"] does not exist, return ? AcquireReadableStreamDefaultReader(this).
-	if options == nil || common.IsNullish(options) || options.Get("mode") == nil || goja.IsUndefined(options.Get("mode")) {
+	if options == nil || common.IsNullish(options) ||
+		options.Get("mode") == nil || sobek.IsUndefined(options.Get("mode")) {
```
Comment (mstoykov) [inline js/modules/k6/experimental/streams/readable_streams.go:68]: This unfortunately becomes 121 symbols and also both cases hit the "is interface nil?" trap so ... yeah. This is the best I can do without having to add a lot mroe code.

**Review by codebien — COMMENTED** [review]
Am I getting it right? Isn't anything else than `s/goja/sobek/`, right?

**Comment by mstoykov** [comment]
Technically it is ... as I am leaving the parser and ast packages from goja in js/compiler. Otherwise it needs a lot more aliases, which won't matter for any other codebase - no extension ever uses those.

And also I replace `goja.` with `sobek.` as to not break a bunch of comments referncing issues and docs for goja.

Both of those will be updated in the future - one of them when we move actually to a fork, the others when we have time as part of cleaning. But I didn't want to figure out all the cases for each comment individually.

**Review by joanlopez — APPROVED** [review]
All the replacements looks good, and I have also confirmed that the only remaining uses of `github.com/dop251/goja` are for `goja/parser` and `goja/file`, expected as described in the issue. 

So, I guess I could say it looks good, as far as I can tell 😄

**Review by codebien — APPROVED** [review]

**Comment by joanlopez** [comment]
Hey! While working on another PR/branch, I realized that for ongoing changes, it's pretty easy to forget about goja imports, as we'll keep the dependency for a while. 

Do you think worths adding a temporary linter that forbids those imports?
cc/ @mstoykov

**Pull request merged**